code/modules/mob/living/damage_procs.dm

```diff
@@ -458,11 +458,12 @@
 	var/old_amount = staminaloss
 	staminaloss = clamp((staminaloss + (amount * CONFIG_GET(number/damage_multiplier))), 0, max_stamina)
 	var/delta = old_amount - staminaloss
-	if(delta <= 0)
-		// need to check for stamcrit AFTER canadjust but BEFORE early return here
-		received_stamina_damage(staminaloss, -1 * delta)
```

Comment:
This comment is here for a reason - I'm pretty sure moving it means that you can't ever refresh stamcrit on someone

Reply:
Because when you're at max staminaloss, `delta` will be equal to `0`. 

You maybe want to move it back to its original position and change it like... 
```dm
if(delta <= 0)
		// need to check for stamcrit AFTER canadjust but BEFORE early return here
		if(amount > 0)
			received_stamina_damage(staminaloss, -1 * delta)
```

Reply:
I tested stamina drain via turrets, disablers, and gym equipment and noticed no difference. Even when stamina was at a partial amount it recovered with no issues.

One piece of code that is relevant is:

```dm
	if(delta == 0) // no change, no need to update
		return 0
```

This check was not happening because `delta <= 0` was always first in the code logic due to the `=` operator. This code also states there is no need to update if delta is 0.

Reply:
I have tested your code and I can say with certainty it breaks stamcrit, preventing people from refreshing it by applying more stam damage.

Reply:
In fact if I would have to hazard a guess I would say I forgot the amount check, considering it's correct in `setStaminaLoss`

![image](https://github.com/user-attachments/assets/7f35d5e7-29a3-4fa9-a7b7-697ec013136e)

Reply:
Fixed via https://github.com/tgstation/tgstation/pull/87810/commits/67299777a4e50cb99c8bc49aa8d62fe41193e7ef